### ISSUE TITLE
setup on Fedora 22: ansible_pkg_mgr should be 'dnf'

### ISSUE THREAD

**Issue by nusenu** [ansible/ansible-modules-core#1797]
Issue Type: Bug Report
Ansible Version: 1.9.2
Summary:

setup module should return dnf (instead of yum) for ansible_pkg_mgr on Fedora 22

Expected Results:
"ansible_pkg_mgr": "dnf"

Actual Results:
"ansible_pkg_mgr": "yum"

**Comment by mscherer** [comment]
Related: https://github.com/ansible/ansible/commit/f624ec4cb8771736ffbe3fe81b2949edda159863

**Comment by knope** [comment]
+1 for dnf support.
  meanwhile... some docs:
https://fedoraproject.org/wiki/Yum_to_DNF_Cheatsheet

and a nifty command provided in the fedora image to merge yum info into dnf, 
(going to run yum module, and then at the end, inform the system the following command via shell.  in the hopes that when the playbook changes, the system wont be borked.):

```
'dnf install python-dnf-plugins-extras-migrate && dnf-2 migrate'
```

###### aside:

this is relevant:
 https://ask.fedoraproject.org/en/question/68046/ansible-dnf-module/
regarding specifically:
ansible -m dnf -a "name=htop" host
it seems installing yum-utils helps... 

`install yum-utils`

###### another aside:

this playbook: https://infrastructure.fedoraproject.org/cgit/ansible.git/tree/playbooks/check-for-updates.yml

it looks like the fedora team works around the issue via issuing both yum and dnf commands... which ... seems bad... but i dont know anything about this (honestly lol).

**Comment by JohnVillalovos** [comment]
I believe a fix has been merged, but won't be seen until Ansible 2.0:

commit f624ec4cb8771736ffbe3fe81b2949edda159863
Author: Michael Scherer misc@zarb.org
Date:   Wed Apr 15 16:11:08 2015 -0400

```
Prefer dnf to yum.

On Fedora 22 and later, yum is deprecated and dnf is installed by
default. However, the detection do not seems to take this in account,
and always use yum, even when yum cli is just a wrapper to tell "use
dnf", as this is the case on F22 and later ( see package dnf-yum ).

As dnf is not installed by default, except on F22, this shouldn't
break anything.
```

**Comment by nusenu** [comment]
I get the following when running
`ansible -m setup fedora23host`:
"ansible_pkg_mgr": "dnf"

with ansible 1.9.4.

thanks!
